sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLV1CompatibilityCheckSuite.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector
+
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.sql.{AnalysisException, QueryTest}
+import org.apache.spark.sql.execution.datasources.v2.V2SessionCatalog
+import org.apache.spark.sql.internal.SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION
+import org.apache.spark.sql.test.SharedSparkSession
+
+class DataSourceV2SQLV1CompatibilityCheckSuite extends QueryTest
+    with SharedSparkSession
+    with BeforeAndAfter {
+
+  before {
+    spark.conf.set(V2_SESSION_CATALOG_IMPLEMENTATION.key, classOf[V2SessionCatalog].getName)
+  }
+
+  after {
+    spark.sessionState.catalog.reset()
+    spark.sessionState.catalogManager.reset()
+    spark.sessionState.conf.clear()
+  }
+
+  test("DeleteFrom: DELETE is only supported with v2 tables") {
```

Comment:
can we put it in `DataSourceV2SQLSuite`? Seems not worthwhile to have a new suite.

Reply:
I tried that but unfortunately fails. 

Below is my understandings (pls correct me if i'm wrong):`DataSourceV2SQLSuite` has set `V2_SESSION_CATALOG_IMPLEMENTATION` to `InMemoryTableSessionCatalog`, but here we need `V2SessionCatalog` which returns a `V1Table` when loading table. We cannot set the current catalog to `V2SessionCatalog` because it doesn't has a register name, and we cannot register `V2SessionCatalog` to a new catalog name because it does not have a constructor without any args. We cannot either set `V2_SESSION_CATALOG_IMPLEMENTATION` to `V2SessionCatalog` because there already has an `V2_SESSION_CATALOG_IMPLEMENTATION` registered.

Reply:
We can follow `DataSourceV2SQLSuite.CreateTableAsSelect: v2 session catalog can load v1 source table`, to unset the v2 session catalog config and use the default impl.

Reply:
Yea, that's exactly what i want, thanks!